olive/common/utils.py

```diff
@@ -27,12 +27,10 @@ def run_subprocess(cmd, env=None, cwd=None, check=False):  # pragma: no cover
         path = env.get("PATH") if env else None
         cmd_exe = shutil.which(cmd[0], path=path)
         cmd[0] = cmd_exe
-    out = subprocess.run(cmd, env=env, cwd=cwd, capture_output=True, check=False)
+    out = subprocess.run(cmd, env=env, cwd=cwd, capture_output=True, check=check)
```

Comment:
Can we keep the old behavior. I made the return status code check separate from this so that we can print the stderr as error message. 

![image](https://github.com/microsoft/Olive/assets/94929125/51e7132d-dff2-42c9-b1fc-5b7954df7717)

Reply:
@jambayk, 
the CalledProcessError.stdout and CalledProcessError.stderro should have the info you want.

Reply:
![image](https://github.com/microsoft/Olive/assets/817030/7bdfce99-ffaf-46b7-bb49-bb3b06c80319)

Reply:
@trajepl as discussed offline, a good middle ground could be we catch the exception like Mike does above, log the stdout and stderr, and rethrow the exception.

Reply:
Updated.